src/transaction.js

```diff
@@ -93,15 +127,23 @@ Transaction.prototype.isCoinbase = function () {
   return this.ins.length === 1 && Transaction.isCoinbaseHash(this.ins[0].hash)
 }
 
-var EMPTY_SCRIPT = new Buffer(0)
-
 Transaction.prototype.addInput = function (hash, index, sequence, scriptSig) {
+  return this._addInput(hash, index, sequence, scriptSig, undefined)
+}
+
+Transaction.prototype.addWitnessInput = function (hash, index, sequence, witness) {
+  return this._addInput(hash, index, sequence, undefined, witness)
+}
```

Comment:
@afk11 I haven't seen that used yet,  so would that literally be an empty P2SH script?  (aka, only the redeemScript, not scriptSig).